### PR TITLE
More world gen mappings

### PR DESCRIPTION
**Pull request by IotaBread**

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]
Looks good from a quick glance.

File: mappings/net/minecraft/util/math/noise/OctavePerlinNoiseSampler.mapping [hunk 1]
```diff
@@ -1,14 +1,37 @@
 CLASS net/minecraft/unmapped/C_hsivpyol net/minecraft/util/math/noise/OctavePerlinNoiseSampler
 	FIELD f_ajyjxtqu octaveSamplers [Lnet/minecraft/unmapped/C_kkpcrsio;
+	FIELD f_enkynfdr firstOctave I
 	FIELD f_kthmwfga lacunarity D
```
Comment (Bubblie01) [inline mappings/net/minecraft/util/math/noise/OctavePerlinNoiseSampler.mapping:4]: does lacunarity mean lake?

File: mappings/net/minecraft/world/biome/source/BiomeAccess.mapping [hunk 2]
```diff
@@ -18,7 +18,7 @@ CLASS net/minecraft/unmapped/C_txmmiamu net/minecraft/world/biome/source/BiomeAc
 	METHOD m_yawvfipv getBiomeForNoiseGen (Lnet/minecraft/unmapped/C_hynzadkk;)Lnet/minecraft/unmapped/C_orlkpefs;
 		ARG 1 pos
 	CLASS C_sutfgxpu Storage
-		METHOD getNoiseBiome (III)Lnet/minecraft/unmapped/C_orlkpefs;
+		METHOD getNoiseBiome getNoiseBiome (III)Lnet/minecraft/unmapped/C_orlkpefs;
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/biome/source/BiomeAccess.mapping:21]: getBiomeNoise maybe?

File: mappings/net/minecraft/world/chunk/Chunk.mapping [hunk 3]
```diff
@@ -5,6 +5,7 @@ CLASS net/minecraft/unmapped/C_lwzmmmqr net/minecraft/world/chunk/Chunk
 	FIELD f_eflzqkft heightLimitView Lnet/minecraft/unmapped/C_qpninoyb;
 	FIELD f_enzfbrzj chunkNoiseSampler Lnet/minecraft/unmapped/C_rvjywqnu;
 	FIELD f_exnoqcnn needsSaving Z
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/chunk/Chunk.mapping:7]: needSaving needsSaving kinda looks confusing imo. This is just personal opinion whatever you choose to keep is fine.

File: mappings/net/minecraft/world/gen/PositionalRandomFactory.mapping [hunk 4]
```diff
@@ -1 +1,13 @@
 CLASS net/minecraft/unmapped/C_coqzjjrt net/minecraft/world/gen/PositionalRandomFactory
+	METHOD m_bpedoycx createRandomAt (Lnet/minecraft/unmapped/C_hynzadkk;)Lnet/minecraft/unmapped/C_rlomrsco;
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/gen/PositionalRandomFactory.mapping:2]: what does this do that the other createRandom methods don't if I can ask?

File: mappings/net/minecraft/world/gen/WorldGenRandom.mapping [hunk 5]
```diff
@@ -1,3 +1,18 @@
 CLASS net/minecraft/unmapped/C_rlomrsco net/minecraft/world/gen/WorldGenRandom
-	METHOD setSeed (J)V
+	METHOD m_bggirwfo createRandomDeriver ()Lnet/minecraft/unmapped/C_coqzjjrt;
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/gen/WorldGenRandom.mapping]: Derivative? createRandomDerive? I don't know what this does so im not entirely sure.

File: mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+		FIELD f_uhernjdb ifTrue Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		METHOD <init> (Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;)V
+			ARG 1 ifTrue
+			ARG 2 thenRun
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+		METHOD equals (Ljava/lang/Object;)Z
+			ARG 1 o
+		METHOD m_kdsbbibg ifTrue ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		METHOD m_szdsyqjr thenRun ()Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping]: thenRun? Idk why but this feels off. Ignore if it is meant to be like this.

File: mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+		FIELD f_mniixoud RULE_CODEC Lcom/mojang/serialization/Codec;
+		FIELD f_uhernjdb ifTrue Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
+		METHOD <init> (Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;Lnet/minecraft/unmapped/C_flergppe$C_dggoeuwn;)V
+			ARG 1 ifTrue
+			ARG 2 thenRun
+		METHOD apply apply (Ljava/lang/Object;)Ljava/lang/Object;
+			ARG 1 context
+		METHOD equals (Ljava/lang/Object;)Z
+			ARG 1 o
+		METHOD m_kdsbbibg ifTrue ()Lnet/minecraft/unmapped/C_flergppe$C_mecsxqth;
```
Comment (Bubblie01) [inline mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping]: isTrue maybe?

**Review by Bubblie01 — COMMENTED** [review]
Most of these are questions and very few suggestions. If you have time please answer them, if you cant answer them just let me know and ill approve this anyway because everything else is looking fine in general. If I know the purpose of what these methods do I can suggest some name changes.

File: mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping (hunk 6 above)
Comment (misode) [reply mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping]: This matches the codec field

File: mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping (hunk 7 above)
Comment (misode) [reply mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping]: Same, `ifTrue` is the codec field. They go together like `ifTrue: { ...}, theRun: { ... }`

File: mappings/net/minecraft/world/chunk/Chunk.mapping (hunk 3 above)
Comment (LambdAurora) [reply mappings/net/minecraft/world/chunk/Chunk.mapping:7]: You say "the chunk need**s** saving", not "the chunk need saving" :P

File: mappings/net/minecraft/world/gen/PositionalRandomFactory.mapping (hunk 4 above)
Comment (IotaBread) [reply mappings/net/minecraft/world/gen/PositionalRandomFactory.mapping:2]: It creates a Random instance using the provided block pos as seed

File: mappings/net/minecraft/world/biome/source/BiomeAccess.mapping (hunk 2 above)
Comment (IotaBread) [reply mappings/net/minecraft/world/biome/source/BiomeAccess.mapping:21]: This is an unobfuscated method

File: mappings/net/minecraft/world/gen/WorldGenRandom.mapping (hunk 5 above)
Comment (IotaBread) [reply mappings/net/minecraft/world/gen/WorldGenRandom.mapping]: Changed it to `createPositionalRandomFactory`

File: mappings/net/minecraft/util/math/noise/OctavePerlinNoiseSampler.mapping (hunk 1 above)
Comment (NoComment1105) [reply mappings/net/minecraft/util/math/noise/OctavePerlinNoiseSampler.mapping:4]: image

**Review by LambdAurora — APPROVED** [review]

**Review by Bubblie01 — APPROVED** [review]
I don't see any issues!

**Pull request merged**